app/controllers/api/base_controller.rb

```diff
@@ -27,8 +27,9 @@ def authenticate
       true
     end
 
-    def current_organization(api_key = nil)
-      @current_organization ||= Organization.find_by(api_key:)
+    def current_organization(value = nil)
+      @current_organization ||=
+        ApiKey.find_by(value:)&.organization || Organization.find_by(api_key: value)
```

Comment:
```suggestion
    def current_organization(api_key = nil)
      @current_organization ||=
        ApiKey.find_by(value: api_key)&.organization || Organization.find_by(api_key:)
```

I would keep `api_key` as name of the signature param as it express what it is at first glance without reading the implementation.